BerkeleyDB TinkerPop test errors in  v7.3.7

The update to BerkeleyDB JE 7.3.7 (https://github.com/JanusGraph/janusgraph/pull/101) causes numerous errors in TinkerPop test suites. To reproduce run the test suite with `test.skip.tp=false`.

```bash
mvn clean install -DskipTests
mvn verify -Dtest.skip.tp=false -pl janusgraph-berkeleyje
```

Failures are shown below.

```
Tests run: 19, Failures: 1, Errors: 17, Skipped: 0, Time elapsed: 8.599 sec <<< FAILURE! - in org.apache.tinkerpop.gremlin.process.traversal.TraversalInterruptionTest
Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 0.001 sec <<< FAILURE! - in org.apache.tinkerpop.gremlin.process.traversal.strategy.decoration.TranslationStrategyProcessTest
Tests run: 24, Failures: 0, Errors: 24, Skipped: 0, Time elapsed: 0.013 sec <<< FAILURE! - in org.apache.tinkerpop.gremlin.process.traversal.strategy.decoration.ElementIdStrategyProcessTest
Tests run: 34, Failures: 0, Errors: 34, Skipped: 0, Time elapsed: 0.047 sec <<< FAILURE! - in org.apache.tinkerpop.gremlin.process.traversal.strategy.decoration.EventStrategyProcessTest
Tests run: 22, Failures: 0, Errors: 22, Skipped: 0, Time elapsed: 0.025 sec <<< FAILURE! - in org.apache.tinkerpop.gremlin.process.traversal.strategy.verification.ReadOnlyStrategyProcessTest
Tests run: 36, Failures: 0, Errors: 36, Skipped: 0, Time elapsed: 0.047 sec <<< FAILURE! - in org.apache.tinkerpop.gremlin.process.traversal.strategy.decoration.PartitionStrategyProcessTest
Tests run: 16, Failures: 0, Errors: 16, Skipped: 0, Time elapsed: 0.025 sec <<< FAILURE! - in org.apache.tinkerpop.gremlin.process.traversal.strategy.decoration.SubgraphStrategyProcessTest
```

The errors have the following form.

```
java.lang.IllegalArgumentException: Could not instantiate implementation: org.janusgraph.diskstorage.berkeleyje.BerkeleyJEStoreManager
        at org.janusgraph.util.system.ConfigurationUtil.instantiate(ConfigurationUtil.java:69)
        at org.janusgraph.diskstorage.Backend.getImplementationClass(Backend.java:480)
        at org.janusgraph.diskstorage.Backend.getStorageManager(Backend.java:414)
        at org.janusgraph.diskstorage.Backend.<init>(Backend.java:148)
        at org.janusgraph.graphdb.JanusGraphBaseTest.clearGraph(JanusGraphBaseTest.java:85)
...
Caused by: org.janusgraph.diskstorage.PermanentBackendException: Error during BerkeleyJE initialization: 
        at org.janusgraph.diskstorage.berkeleyje.BerkeleyJEStoreManager.initialize(BerkeleyJEStoreManager.java:126)
        at org.janusgraph.diskstorage.berkeleyje.BerkeleyJEStoreManager.<init>(BerkeleyJEStoreManager.java:82)
...
Caused by: com.sleepycat.je.ThreadInterruptedException: (JE 7.3.7) Environment must be closed, caused by: com.sleepycat.je.ThreadInterruptedException: Environment invalid because of previous exception: (JE 7.3.7) target/db/standard java.lang.InterruptedException THREAD_INTERRUPTED: InterruptedException may cause incorrect internal state, unable to continue. Environment is invalid and must be closed.
        at com.sleepycat.je.ThreadInterruptedException.wrapSelf(ThreadInterruptedException.java:105)
        at com.sleepycat.je.dbi.EnvironmentImpl.checkIfInvalid(EnvironmentImpl.java:1766)
        at com.sleepycat.je.dbi.DbEnvPool.getEnvironment(DbEnvPool.java:151)
        at com.sleepycat.je.Environment.makeEnvironmentImpl(Environment.java:267)
        at com.sleepycat.je.Environment.<init>(Environment.java:252)
        at com.sleepycat.je.Environment.<init>(Environment.java:222)
        at org.janusgraph.diskstorage.berkeleyje.BerkeleyJEStoreManager.initialize(BerkeleyJEStoreManager.java:122)
        at org.janusgraph.diskstorage.berkeleyje.BerkeleyJEStoreManager.<init>(BerkeleyJEStoreManager.java:82)
...
Caused by: com.sleepycat.je.ThreadInterruptedException: Environment invalid because of previous exception: (JE 7.3.7) target/db/standard java.lang.InterruptedException THREAD_INTERRUPTED: InterruptedException may cause incorrect internal state, unable to continue. Environment is invalid and must be closed.
        at com.sleepycat.je.latch.SharedLatchImpl.acquireShared(SharedLatchImpl.java:109)
        at com.sleepycat.je.txn.TxnManager.registerTxn(TxnManager.java:213)
        at com.sleepycat.je.txn.Txn.initTxn(Txn.java:392)
        at com.sleepycat.je.txn.Txn.<init>(Txn.java:286)
        at com.sleepycat.je.txn.Txn.<init>(Txn.java:265)
        at com.sleepycat.je.txn.Txn.createLocalTxn(Txn.java:292)
        at com.sleepycat.je.txn.Txn.createUserTxn(Txn.java:314)
        at com.sleepycat.je.txn.TxnManager.txnBegin(TxnManager.java:199)
        at com.sleepycat.je.dbi.EnvironmentImpl.txnBegin(EnvironmentImpl.java:2460)
        at com.sleepycat.je.Environment.beginTransactionInternal(Environment.java:1482)
        at com.sleepycat.je.Environment.beginTransaction(Environment.java:1367)
        at org.janusgraph.diskstorage.berkeleyje.BerkeleyJEStoreManager.beginTransaction(BerkeleyJEStoreManager.java:152)
```

--- a/janusgraph-berkeleyje/src/test/java/org/janusgraph/blueprints/process/BerkeleyJanusGraphProcessTest.java
+++ b/janusgraph-berkeleyje/src/test/java/org/janusgraph/blueprints/process/BerkeleyJanusGraphProcessTest.java
@@ -23,7 +23,7 @@ import org.junit.runner.RunWith;
 /**
  * @author Matthias Broecheler (me@matthiasb.com)
  */
-@RunWith(ProcessStandardSuite.class)
+@RunWith(BerkeleyProcessStandardSuite.class)
 @GraphProviderClass(provider = BerkeleyGraphProvider.class, graph = JanusGraph.class)
 public class BerkeleyJanusGraphProcessTest {
 }

--- a/janusgraph-berkeleyje/src/test/java/org/janusgraph/blueprints/process/BerkeleyProcessStandardSuite.java
+++ b/janusgraph-berkeleyje/src/test/java/org/janusgraph/blueprints/process/BerkeleyProcessStandardSuite.java
@@ -1,0 +1,44 @@
+// Copyright 2017 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.blueprints.process;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.tinkerpop.gremlin.process.ProcessStandardSuite;
+import org.apache.tinkerpop.gremlin.process.traversal.TraversalInterruptionTest;
+import org.junit.runners.model.InitializationError;
+import org.junit.runners.model.RunnerBuilder;
+
+import java.lang.reflect.Field;
+
+/**
+ * Custom TinkerPop {@link ProcessStandardSuite} that excludes {@link TraversalInterruptionTest} for compatibility with
+ * BerkeleyDB JE, which does not support thread interrupts.
+ */
+public class BerkeleyProcessStandardSuite extends ProcessStandardSuite {
+
+    public BerkeleyProcessStandardSuite(final Class<?> klass, final RunnerBuilder builder) throws InitializationError {
+        super(klass, builder, getTestList());
+    }
+
+    private static final Class<?>[] getTestList() throws InitializationError {
+        try {
+            final Field field = ProcessStandardSuite.class.getDeclaredField("allTests");
+            field.setAccessible(true);
+            return (Class<?>[]) ArrayUtils.removeElement((Class<?>[]) field.get(null), TraversalInterruptionTest.class);
+        } catch (ReflectiveOperationException e) {
+            throw new InitializationError("Unable to create test list");
+        }
+    }
+}

